include/jemalloc/internal/seq.h

```diff
@@ -0,0 +1,55 @@
+#ifndef JEMALLOC_INTERNAL_SEQ_H
+#define JEMALLOC_INTERNAL_SEQ_H
+
+#include "jemalloc/internal/atomic.h"
+
+/*
+ * A simple seqlock implementation.
+ */
+
+#define seq_define(type, short_type)					\
+typedef struct {							\
+	atomic_zu_t seq;						\
+	atomic_zu_t data[						\
```

Comment:
Yeah -- it's UB to have races on non-atomics. (Well, OK, we ignore that sometimes because of optimizer bugs, but we should probably try not to do so if it's not necessary).